[DataLoader] Fix  collation logic

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97789

Similar to #97737, a previous auto-refactor changed how `bytes` are handled during collation, which can potentially lead to performance regression. This PR undoes that.

NivekT committed:
[DataLoader] Fix  collation logic

[ghstack-poisoned]

albanD reviewed (approved):
Thanks!
Should we have a one off test for this?

FYI @malfet

NivekT commented:
@pytorchbot merge -r

NivekT committed:
Update on "[DataLoader] Fix  collation logic"


Similar to #97737, a previous auto-refactor changed how `bytes` are handled during collation, which can potentially lead to performance regression. This PR undoes that.

[ghstack-poisoned]
